spk/kavita/Makefile

```diff
@@ -0,0 +1,43 @@
+SPK_NAME = kavita
+SPK_VERS = 0.7.12
+SPK_REV = 1
+SPK_ICON = src/kavita.png
+
+DEPENDS = cross/kavita
+
+# Arch exclusions for dotnet 6.0
+DOTNET_CORE_ARCHS = 1
+
+# Requires glibc 2.28 which is not in DSM 6 toolchains
+REQUIRED_MIN_DSM = 7.0
+
+MAINTAINER = SynoCommunity
+MAINTAINER_URL = https://synocommunity.com/
+DESCRIPTION  = Kavita is a fast, feature rich, cross platform reading server. Built with a focus for being a full solution for all your reading needs.
+CHANGELOG = "Initial release"
+DISPLAY_NAME = Kavita
+HOMEPAGE = https://www.kavitareader.com/
+LICENSE  = GPLv3
+LICENSE_FILE = $(WORK_DIR)/Kavita/LICENSE.txt
+HELPURL = https://wiki.kavitareader.com/
+
+STARTABLE = yes
+SERVICE_USER = auto
+SERVICE_SETUP = src/service-setup.sh
+SERVICE_PORT = 5500
+SERVICE_PORT_TITLE = $(DISPLAY_NAME)
+
+# Admin link for in DSM UI
+ADMIN_PORT = $(SERVICE_PORT)
+
+WIZARDS_DIR = src/wizard/
+
+POST_STRIP_TARGET = kavita_extra_install
+
+include ../../mk/spksrc.spk.mk
+
+.PHONY: kavita_extra_install
+kavita_extra_install:
+	@$(MSG) "Install appsettings.json."
+	@install -m 755 -d $(STAGING_DIR)/var/config
+	@install -m 644 src/appsettings.json $(STAGING_DIR)/var/config/appsettings.json
```

Comment:
why not installing appsettings.json to the var folder?
this would avoid the need of awkward link

BTW: isn't there a parameter to tell kavita where to find the appsettings?

Reply:
I wanted to ensure that the config folder is stored away from the binaries since it is the only folder that must be retained on upgrade. I would have preferred to use a launch parameter but I haven't been able to find any defined in the source documentation for Kavita.

Reply:
Actually I thought I was placing the config folder in the var folder. Based on other examples I see I thought this was the way to do it.

Reply:
@hgy59, I reached out to the Kavita developers and they confirmed that they don't have configurable parameters for the location of the config folder. One said:

> no, Kavita doesn't allow moving the config folder. Having it in the install folder is ideal to prevent users from editing anything and ensuring it's always there

When asked if this could be a future feature request (given that hard/soft links can be messy for updates), they replied:

> Unfortunately that's not something I'd be open to.